plugins/test/dynamic_test.cc

```diff
@@ -384,6 +521,25 @@ void DynamicTest::BenchHttpHandlers(benchmark::State& state) {
       PrepBodyCallbackBenchmark(cfg_, cfg_.response_body());
   BM_RETURN_IF_ERROR(response_body_chunks.status());
 
+  std::vector<BackgroundStream> background_streams;
+  if (env_.num_background_streams() > 0) {
+    for (int i = 0; i < env_.num_background_streams(); ++i) {
+      background_streams.emplace_back(
+          handle, request_headers.has_value() ? &*request_headers : nullptr,
+          response_headers.has_value() ? &*response_headers : nullptr,
+          &*request_body_chunks, &*response_body_chunks);
+      // Advance twice, once to create the TestHttpContext and once to perform
+      // an HTTP handler callback.
+      BM_RETURN_IF_ERROR(background_streams.back().Advance());
+      BM_RETURN_IF_ERROR(background_streams.back().Advance());
```

Comment:
Is the second Advance() really needed here? I assume the first HTTP handler could also be invoked below in the main loop.

Reply:
While it's probably the case that the benchmark loop would execute enough times to advance all streams to the point where they actually perform an HTTP callback and incur whatever memory usage comes with it, this ensures that, removing some (slight) unpredictability. While I agree it's a little clunky aesthetically, I don't see any other downside, so would prefer to let function take precedence over form here.